artiq/coredevice/suservo.py

```diff
@@ -564,3 +607,254 @@ def set_y(self, profile, y):
             raise ValueError("Invalid SUServo y-value!")
         self.set_y_mu(profile, y_mu)
         return y_mu
+
+
+class _MaskedIOUpdate:
+    def __init__(self, core, cpld, dds, io_update):
+        self.cpld = cpld
+        self.dds = dds
+        self.core = core
+        self.io_update = io_update
+
+        # Synchronized SU-Servo uses the designated I/O update pin.
+        # When communicating with the DDS via slow SPI, MASK_NU must be set to
+        # perform serial transfer, then unset to propagate I/O update.
+        self.toggle_mask_nu = self.cpld.io_update is not None
+    
+    @kernel
+    def pulse_mu(self, duration):
+        """Unset MASK_NU, then pulse the IO Update TTL high for the specified
+        duration (in machine units). MASK_NU is restored to the previous value
+        after the I/O pulse.
+
+        The I/O update TTL supports fine timestamp.
+        The preamble CFG write advances the timestamp to a fine-timestamp
+        invariant position, then I/O update TTL pulses.
```

Comment:
"fine-timestamp invariant position" is rather confusing. How about "aligned to the coarse RTIO clock"?